File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.js
function solution(code) {
    let mode = false;
    let ret = '';
    
    for (let i = 0; i < code.length; i++) {
        if (!mode) {
            if (code[i] === '1') {
                mode = !mode;
                continue;
            }
            
            if (i % 2 === 0) {
                ret += code[i];
            }
        } else {
            if (code[i] === '1') {
                mode = !mode;
                continue;
            }
            
            if (i % 2 === 1) {
                ret += code[i];
            }
        }
    }
    
    return ret.length === 0 ? 'EMPTY' : ret;
}

File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String ret = "";
        int mode = 0;
        
        for (int i = 0; i < code.length(); i++) {
            char ch = code.charAt(i);
            
            if (mode == 0) {
                if (ch == '1') {
                    mode = 1;
                } else if (i % 2 == 0) {
                    ret += ch;
                }
            } else {
                if (ch == '1') {
                    mode = 0;
                } else if (i % 2 == 1) {
                    ret += ch;
                }
            }
        }
        
        if (ret.length() == 0) {
            return "EMPTY";
        }
        
        return ret;
    }
}